### DISCUSSION TITLE
functions returning 64 bit integers

### DISCUSSION THREAD

**Discussion by sbraccia** [cunnane/xloil#61] [Q&A]
First of all thank you for this great project.

I was wondering if it is expected for the simple function below to return -1 (both python and excel are 64 bit)?

@xloil.func()
def test():
    return 2147483647 + 1

The same function if defined as async would throw an OverflowError

Many thanks,
Carlo

**Comment by cunnane** [comment]
Thank you and I'm glad you're enjoying xloil.

Excel natively only supports 32 bit integers and double precision floating
point.  You could return very large integers to Excel via the object cache
functionality, but then you would not be able to use Excel's arithmetic
functions on them, you'd need to define your own SUM etc.  Since python is
dynamic you could define something which  worked like `=pyExec("2*a+b**2",
A1, A2)` which would work if A1 or A2 were very-large-integer objects.

**Reply by sbraccia** [reply]
That explains the behaviour, thank you. For my use case casting to a float works just fine.
Thank you again
Kind regards

**Comment by cunnane** [comment]
I'll add a warning or error if big ints are encountered as the current
behaviour is a bit surprising!
